src/workerd/jsg/jsg.h

```diff
@@ -303,6 +303,36 @@ class JsExceptionThrown: public std::exception {
 //
 //     JSG_METHOD_NAMED(delete, delete_);
 
+#define JSG_METHOD_UNBOUND(name) \
+  do { \
+    static const char NAME[] = #name; \
+    registry.template registerMethodWithoutSignature<NAME, decltype(Self::name), &Self::name>(); \
+  } while (false)
+// Like JSG_METHOD except that while the method is attached to the prototype, it is not bound
+// to the `this`. This specifically allows for using such methods in JavaScript using destructuring
+// assignment, for instance:
+//   const { foo } = object;
+//   foo();
+//
+// With JSG_METHOD, such a method call would error with an "illegal invocation" error.
+//
+// Such methods MUST be static on the resource type.
+
+#define JSG_METHOD_UNBOUND_NAMED(name, method) \
+  do { \
+    static const char NAME[] = #name; \
+    registry.template registerMethodWithoutSignature<NAME, decltype(Self::name), &Self::name>(); \
```

Comment:
```suggestion
    registry.template registerMethodWithoutSignature<NAME, decltype(Self::method), &Self::method>(); \
```

We should have test cases for these new macros, too.

Reply:
These are going to be removed from this PR since I am removing the numeric ID APIs here.